application/indexer/value.go

```diff
@@ -93,53 +99,74 @@ func (vi *ValueIndex) Drop(txn *badger.Txn, utxoID []byte) error {
 	return utils.DeleteValue(txn, key)
 }
 
-func (vi *ValueIndex) GetValueForOwner(txn *badger.Txn, owner *objs.Owner, minValue *uint256.Uint256, exclude [][]byte) ([][]byte, *uint256.Uint256, error) {
-	exclusionSet := make(map[[constants.HashLen]byte]bool)
-	var hsh [constants.HashLen]byte
-	for j := 0; j < len(exclude); j++ {
-		ID := utils.CopySlice(exclude[j])
-		copy(hsh[:], utils.CopySlice(ID))
-		exclusionSet[hsh] = true
-	}
-	result := [][]byte{}
-	valueCount := uint256.Zero()
-	opts := badger.DefaultIteratorOptions
-	prefix := vi.prefix()
+func (vi *ValueIndex) GetValueForOwner(txn *badger.Txn, owner *objs.Owner, minValue *uint256.Uint256, excludeFn func([]byte) (bool, error), maxCount int, lastKey []byte) ([][]byte, *uint256.Uint256, []byte, error) {
 	ownerBytes, err := owner.MarshalBinary()
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
+
+	prefix := vi.prefix()
 	prefix = append(prefix, ownerBytes...)
-	prefixLen := len(prefix)
+	opts := badger.DefaultIteratorOptions
 	opts.Prefix = prefix
 	iter := txn.NewIterator(opts)
 	defer iter.Close()
-	for iter.Seek(prefix); iter.ValidForPrefix(prefix); iter.Next() {
+
+	result := [][]byte{}
+	totalValue := uint256.Zero()
+	prefixLen := len(prefix)
+
+	if lastKey != nil {
```

Comment:
Simplify / left-align?

seekKey := prefix
if lastKey != nil {
  seekKey = lastKey
  // prevents need to call next before iteration
  seekKey[len(seekKey)-1]+=1
}
for iter.Seek(seekKey); iter.ValidForPrefix(prefix); iter.Next() {

Reply:
So it's a bit more complex. In case we do have a lastKey, we first seek to that key, and then we seek to the next one (because that last key has already been processed), but before we can do that we first have to check if that lastkey was valid at all (since its user input, it can be tampered with), otherwise we risk a panic.

So essentially the case of lastKey ends with a `iter.Next` call while the other case ends with an `iter.Seek` call, so I couldn't generalize much of it.